arch/x86/32/common/source/ArchThreads.cpp

```diff
@@ -23,9 +23,8 @@ void ArchThreads::setAddressSpace(Thread *thread, ArchMemory& arch_memory)
 
 void ArchThreads::createBaseThreadRegisters(ArchThreadRegisters *&info, void* start_function, void* stack)
 {
-  info = (ArchThreadRegisters*)new uint8[sizeof(ArchThreadRegisters)];
-  memset((void*)info, 0, sizeof(ArchThreadRegisters));
-  pointer root_of_kernel_paging_structure = VIRTUAL_TO_PHYSICAL_BOOT(((pointer)ArchMemory::getRootOfKernelPagingStructure()));
+  info = new ArchThreadRegisters();
```

Comment:
is this guaranteed to null the members?

Reply:
Yes. ```new ArchThreadRegisters()``` value-initializes members, which will set all built-in types to 0.
Just ```new ArchThreadRegisters``` without the parentheses wouldn't initialize the struct members.